src/python/pants/backend/docker/util_rules/docker_binary_test.py

```diff
@@ -87,3 +96,52 @@ def test_docker_binary_run_image(docker_path: str, docker: DockerBinary) -> None
         description="",  # The description field is marked `compare=False`
     )
     assert run_request.description == f"Running docker image {image_ref}"
+
+
+from pants.core.util_rules.system_binaries import (
+    BinaryPath,
+    BinaryPathRequest,
+    BinaryPaths,
+    BinaryShims,
+    BinaryShimsRequest,
+)
+
+
+@pytest.mark.parametrize("podman_enabled", [True, False])
+@pytest.mark.parametrize("podman_found", [True, False])
+def test_get_docker(rule_runner: RuleRunner, podman_enabled, podman_found) -> None:
+    docker_options = create_subsystem(DockerOptions, enable_podman=podman_enabled, tools=[])
+    docker_options_env_aware = mock.MagicMock(spec=DockerOptions.EnvironmentAware)
```

Comment:
Is this the correct way to mock this out? I couldn't find a single existing case of a test needing to mock an `EnvironmentAware` class, and it's all a bit _magic_ when trying to work out how to fake it.

Reply:
not sure about this 🤷🏽 
if the tests say OK, then it looks good to me 😬